app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.java

```diff
@@ -0,0 +1,116 @@
+package org.kiwix.kiwixmobile.webserver;
+
+import android.util.Log;
+import androidx.annotation.NonNull;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import org.kiwix.kiwixlib.JNIKiwixException;
+import org.kiwix.kiwixlib.JNIKiwixLibrary;
+import org.kiwix.kiwixlib.JNIKiwixServer;
+
+/**
+ * WebServerHelper class is used to set up the suitable environment i.e. getting the
+ * ip address and port no. before starting the WebServer
+ * Created by Adeel Zafar on 18/07/2019.
+ */
+
+public class WebServerHelper {
+  public static boolean isServerStarted;
+  private static int port;
+  private final JNIKiwixLibrary kiwixLibrary = new JNIKiwixLibrary();
+  private final JNIKiwixServer kiwixServer = new JNIKiwixServer(kiwixLibrary);
+  private static final String TAG = "WebServerHelper";
+
+  public WebServerHelper() {
+
+  }
+
+  public boolean startServerHelper(@NonNull ServerStateListener stateListener,
+      @NonNull ArrayList<String> selectedBooksPath) {
+
+    // 1. Get port from settings screen
+    // 2. Ask user to change port in settings if port is in use.
+    // OR
+    // Always use 8080 and when its not available then iterate this number.
+    String ip = getIpAddress();
+    ip = ip.replaceAll("\n", "");
+    if (ip.length() == 0) {
+      stateListener.serverFailed();
+    } else if (!isServerStarted && startAndroidWebServer(selectedBooksPath)) {
+      stateListener.serverStarted("http://" + ip + ":" + port);
+    }
+    return isServerStarted;
+  }
+
+  public boolean stopAndroidWebServer(@NonNull ServerStateListener stateListener) {
+    if (isServerStarted) {
+      kiwixServer.stop();
+      isServerStarted = false;
+      stateListener.serverStopped();
+      return true;
+    }
+    return false;
+  }
+
+  private boolean startAndroidWebServer(ArrayList<String> selectedBooksPath) {
+    if (!isServerStarted) {
+      port = 8080;
```

Comment:
extract 8080 to a constant `ctrl+alt+c` named `DEFAULT_PORT`

Reply:
Fixed. 04cf0398c6b24814ba458ee6560506595c4ac139